### PR TITLE
Fix rails 5 deprication warnings

### PR DESCRIPTION
**Pull request by olegantonyan**
Fix these annoying deprecation warnings:
```
DEPRECATION WARNING: before_filter is deprecated and will be removed in Rails 5.1. Use before_action instead. (called from get at /home/oleg/.rbenv/versions/2.3.0/lib64/ruby/2.3.0/forwardable.rb:184)
DEPRECATION WARNING: env is deprecated and will be removed from Rails 5.1. (called from get at /home/oleg/.rbenv/versions/2.3.0/lib64/ruby/2.3.0/forwardable.rb:184)
```

### CODE REVIEW

File: lib/gaffe/errors.rb [hunk 1]
```diff
@@ -3,8 +3,13 @@ module Errors
...
     included do
-      before_filter :fetch_exception, only: %w(show)
-      before_filter :append_view_paths
+      if Rails::VERSION::MAJOR >= 5
+        before_action :fetch_exception, only: %w(show)
+        before_action :append_view_paths
+      else
+        before_filter :fetch_exception, only: %w(show)
+        before_filter :append_view_paths
+      end
```
Comment (remiprev) [inline lib/gaffe/errors.rb]: I think we can just do:

```ruby
before_action :fetch_exception, only: %w(show)
before_action :append_view_paths
```

`before_action` is supported in Rails 4 :smile:

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -27,6 +27,19 @@
 
 # We need a fake "ApplicationController" because Gaffe's default controller inherits from it
 class ApplicationController < ActionController::Base
+  if Rails::VERSION::MAJOR < 4
+    def self.before_action(*args)
+      before_filter(*args)
+    end
+  end
```
Comment (remiprev) [inline spec/spec_helper.rb]: Same thing here, `before_action` should work correctly in Rails 4’s `ActionController::Base`.

File: lib/gaffe/version.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 module Gaffe
-  VERSION = '1.0.2'
+  VERSION = '1.0.2'.freeze
```
Comment (remiprev) [inline lib/gaffe/version.rb]: This is not related to the deprecation notices, is it?

File: lib/gaffe/errors.rb (hunk 1 above)
Comment (olegantonyan) [inline lib/gaffe/errors.rb]: Yes, but it is not in rails 3 which is also supported by gaffe

Comment (remiprev) [inline lib/gaffe/errors.rb]: Oh, I think this might be a good time to drop support for Rails 3 (and 1.9.3) then. Rails 3 is not supported by the Rails team anymore.

File: lib/gaffe/version.rb (hunk 3 above)
Comment (olegantonyan) [inline lib/gaffe/version.rb]: Yes, but this makes rubocop happy. Otherwise travis build fails because of rubocop

**Comment by olegantonyan** [comment]
So, what about this? Will we merge it or drop rails3 support first?

**Comment by remiprev** [comment]
I think we should drop Rails 3 support for the next release. Is that something you might be interested in doing?

**Comment by olegantonyan** [comment]
Well, I can try
